week2/project/ecommerce2/src/components/ProductDetails.js

```diff
@@ -4,7 +4,7 @@ import ErrorMsg from './Error';
 
 export default function ProductDetails() {
   const { id } = useParams();
-  const [product, setProduct] = useState([]);
+  const [product, setProduct] = useState(0);
```

Comment:
yes now the product is `0` or usually we set it as `null` however, you need to render the product element to the user when only you get the product object right ?

because right now I get an error that there is no properties of undefined
so you need to go to the element and make a logic for it 

```js
{product && !isLoading && ...render the element}
```